scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -0,0 +1,208 @@
+import com.esri.arcgisruntime.mapping.view.ImageOverlay;
+import com.esri.arcgisruntime.mapping.view.SceneView;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Slider;
+
+public class AnimateImagesWithImageOverlayController {
+
+  @FXML
+  private SceneView sceneView;
+  @FXML
+  private Button controlAnimationButton;
+  @FXML
+  private Slider opacitySlider;
+  @FXML
+  private ComboBox<String> framesComboBox;
+
+  private List<ImageFrame> imageFrames;
+
+  private Integer imageIndex = 0;
+  private Integer period = 67;
+
+  private Timer timer = null;
+  private boolean isTimerRunning = true;
+
+  public void initialize() {
+
+    try {
+
+      // create a new ArcGISScene and set it to the scene view
+      ArcGISScene scene = new ArcGISScene();
+      sceneView.setArcGISScene(scene);
+
+      // create a camera, looking at the pacific southwest sector
+      Point observationPoint = new Point(-116.621, 24.7773, 856977.0);
+      Camera camera = new Camera(observationPoint, 353.994, 48.5495, 0.0);
+
+      // create an envelope of the pacific southwest sector for displaying the image frame
+      Point pointForImageFrame = new Point(-120.0724273439448, 35.131016955536694, SpatialReferences.getWgs84());
+      Envelope imageFrameEnvelope = new Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488);
+      scene.setInitialViewpoint(new Viewpoint(imageFrameEnvelope, camera));
+
+      // create a new tiled layer from the World Dark Gray Base REST service and set it as the scene's basemap
+      Basemap basemap = new Basemap(new ArcGISTiledLayer("https://services.arcgisonline" +
+        ".com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer"));
+      scene.setBasemap(basemap);
+
+      // create a new elevation source from the Terrain3D REST service and set it as the scene's base surface
+      Surface surface = new Surface();
+      surface.getElevationSources().add(new ArcGISTiledElevationSource("https://elevation3d.arcgis" +
+        ".com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"));
+      scene.setBaseSurface(surface);
+
+      // create and append an image overlay to the scene view
+      sceneView.getImageOverlays().add(new ImageOverlay());
+
+      // instantiate a new empty list to hold image frames
+      imageFrames = new ArrayList<>();
+      // get the image files from local storage as an unordered list
+      File[] imageFiles = new File(System.getProperty("data.dir"), "./samples-data/PacificSouthWest").listFiles();
+      // sort the list of image files
+      if (imageFiles != null) {
+        Arrays.sort(imageFiles);
+        // create an image with the given path and use it to create an image frame
+        for (File file : imageFiles) {
+          ImageFrame imageFrame = new ImageFrame(file.getAbsolutePath(), imageFrameEnvelope);
+          imageFrames.add(imageFrame);
+        }
+      }
```

Comment:
Some suggestions for the comments in this section. We should specify how we're sorting the file paths and ideall why (so that they display in order). Did you try without sorting? I would have thought the files are read in alphabetically when you do `.listFiles()`.
In the last section we're not actually creating an image in the intermediate step, we're using the `ImageFrame(URI, Envelope) constructor, so we should specify that there. Perhaps it's worth adding a comment somewhere that we can do `ImageFrame(Image, Envelope)`

```suggestion
      // instantiate a new empty list to hold image frames
      imageFrames = new ArrayList<>();
      
      // get the image files from local storage
      File[] imageFiles = new File(System.getProperty("data.dir"), "./samples-data/PacificSouthWest").listFiles();
      
      // sort the list of image files by file name in ascending order
      if (imageFiles != null) {
        Arrays.sort(imageFiles);
        
        // create an image frame with the file path URI and add it to the list of image frames
        for (File file : imageFiles) {
          ImageFrame imageFrame = new ImageFrame(file.getAbsolutePath(), imageFrameEnvelope);
          imageFrames.add(imageFrame);
        }
      }
```

Reply:
Great thanks for the comment improvement suggestions. And yep, so we do need to sort to be on the safe side - here's the doc from the `.listFiles()`:

> There is no guarantee that the name strings in the resulting array will appear in any specific order; they are not, in particular,guaranteed to appear in alphabetical order.

Reply:
Good find!